Crypto101.org

```diff
@@ -2912,7 +2912,40 @@ http://www.cs.ucdavis.edu/~rogaway/papers/relates.pdf for further reading
 
 *** MD5
 
-TODO: Explain MD5
+MD5 is a hash function designed by Ronald Rivest in 1991 as an extension of MD4. This hash function outputs 128-bit digests. Over the course of the years, the cryptographic community has repeatedly uncovered MD5's weaknesses. In 1993, Bert den Boer and Antoon Bosselaers published a paper demonstrating "pseudo-collisions" for the compression function of MD5. Dobbertin expanded upon this research and was able to produce collisions for the compression function. In 2004, based on Dobbertin's work, Xiaoyun Wang, Dengguo Feng, Xuejia Lai and Hongbo Yu showed that MD5 is vulnerable to real collision attacks. The last straw came when Xiaoyun Wang et al. managed to generate colliding X.509 certificates and then presented a distinguishing attack on HMAC-MD5.
```

Comment:
Without an in-text citation, the citations won't show up at the back of the rendered book even if they're in the BiBTeX. (Most academics maintain a giant BiBTeX file somewhere that gradually accretes citations; they're not necessarily per-document.)

You'd do this as `\cite{cryptoeprint:2005:067}`; grep the source for `\cite` for some style suggestions.